analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/mvn/api/impl/VulnerableCallChainsApiServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.analyzer.restapiplugin.mvn.api.impl;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import eu.fasten.analyzer.restapiplugin.mvn.KnowledgeBaseConnector;
+import eu.fasten.analyzer.restapiplugin.mvn.api.VulnerableCallChainsApiService;
+import eu.fasten.core.data.FastenURI;
+import eu.fasten.core.vulchains.VulnerableCallChain;
+import eu.fasten.core.vulchains.VulnerableCallChainRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+
+
+@Lazy
+@Service
+public class VulnerableCallChainsApiServiceImpl implements VulnerableCallChainsApiService {
+
+    private static final Logger logger = LoggerFactory.getLogger(VulnerableCallChainsApiServiceImpl.class);
+
+    private VulnerableCallChainRepository vulnerableCallChainRepository;
+
+    public VulnerableCallChainsApiServiceImpl() {
+        try {
+            vulnerableCallChainRepository = new VulnerableCallChainRepository(KnowledgeBaseConnector.vulnerableCallChainsPath);
+        } catch (Exception e) {
+            logger.error("Error constructing Vulnerability Call Chain Repository", e);
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Helper method to convert set of vulnerable call chains to JSON formatted string response.
+     * @param chains - a set of {@link VulnerableCallChain} objects to be serialized.
+     * @return {@link String} formatted as JSON response.
+     */
+    private String VulnerableCallChainsToJSON(Set<VulnerableCallChain> chains) {
+        GsonBuilder builder = new GsonBuilder();
```

Comment:
We have `eu.fasten.core.vulchains.VulnerableCallChainJsonUtils` that you can use for the serialization and deserialization. I would suggest using that for the sake of consistency.

Reply:
> mnt/fasten/vul-path

Sure. It's more of a change for k8s-deployments repo. I changed it. You may also want to have a look at PR there.

> eu.fasten.core.vulchains.VulnerableCallChainJsonUtils

Well, the format you store as json in your files is different what we want to return to client. Do you remember the thing about different types of fasten uri? So, it still doesn't work as expected. It still returns just a raw uri in json, and not the object with parsed properties.

Although I don't think you need it such in your implementations. So maybe we can just to create a different json util for my case, rather than adjusting yours?

Reply:
So, to remind you the difference,

<img width="769" alt="Screenshot 2022-01-17 at 12 59 03" src="https://user-images.githubusercontent.com/14923964/149765956-efa845c5-9ba1-4fdd-bb8f-d721258273bf.png">
<img width="1151" alt="Screenshot 2022-01-17 at 13 00 16" src="https://user-images.githubusercontent.com/14923964/149765963-ebf10964-624c-4774-b906-a457575f6d43.png">

The first is what I need on the client, the second is what returned by your json util.